react/features/desktop-picker/components/DesktopPicker.js

```diff
@@ -0,0 +1,257 @@
+     * Sets a default selected source if one is not already set.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only React Component props that this
+     * instance will receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (!this.state.selectedSourceId
+            && nextProps.sources.screen.length) {
+            this.setState({ selectedSourceId: nextProps.sources.screen[0].id });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <Dialog
+                isModal = { false }
+                okTitleKey = 'dialog.Share'
+                onCancel = { this._onCloseModal }
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.shareYourScreen'
+                width = 'medium' >
+                <Tabs tabs = { this._renderTabs() } />
+            </Dialog>);
+    }
+
+    /**
+     * Dispatches an action to get currently available DesktopCapturerSources.
+     *
+     * @private
+     * @returns {void}
+     */
+    _updateSources() {
+        this.props.dispatch(obtainDesktopSources(
+            typesToFetch,
+            {
+                thumbnailSize: config.desktopSharingElectronThumbnailSize
+            }
+        ));
+    }
+
+    /**
+     * Create an interval to update knwon available DesktopCapturerSources.
+     *
+     * @private
+     * @returns {void}
+     */
+    _startPolling() {
+        this._stopPolling();
+        this._poller = window.setInterval(this._updateSources,
+            updateInterval);
+    }
+
+    /**
+     * Cancels the interval to update DesktopCapturerSources.
+     *
+     * @private
+     * @returns {void}
+     */
+    _stopPolling() {
+        window.clearInterval(this._poller);
+    }
+
+    /**
+     * Sets the currently selected DesktopCapturerSource.
+     *
+     * @param {string} id - The id of DesktopCapturerSource.
+     * @returns {void}
+     */
+    _onPreviewClick(id) {
+        this.setState({ selectedSourceId: id });
+    }
+
+    /**
+     * Request to close the modal and execute callbacks
+     * with the selected source id.
+     *
+     * @returns {void}
+     */
+    _onSubmit() {
+        this._onCloseModal(this.state.selectedSourceId);
+    }
+
+    /**
+     * Dispatches an action to hide the DesktopPicker and invokes
+     * the passed in callback with a selectedSourceId, if any.
+     *
+     * @param {string} id - The id of the DesktopCapturerSource to pass into
+     * the onSourceChoose callback.
+     * @returns {void}
+     */
+    _onCloseModal(id = '') {
+        this.props.onSourceChoose(id);
+        this.props.dispatch(hideDialog());
+    }
+
+    /**
+     * Configures and renders the tabs for display.
+     *
+     * @returns {ReactElement}
+     * @private
+     */
+    _renderTabs() {
```

Comment:
The function name is a bit misleading here, since it doesn't render anything, it just returns some objects. Maybe have it return the Tabs component